src/app/shared/components/workshop-card/workshop-card.component.ts

```diff
@@ -108,6 +110,25 @@ export class WorkshopCardComponent implements OnInit, OnDestroy {
     this.isFavorite = !this.isFavorite;
   }
 
+  onChange(): void {
+    const dialogRef = this.dialog.open(ConfirmationModalWindowComponent, {
+      width: Constants.MODAL_SMALL,
+      data: {
+        type: this.workshopData.status === 'Open' ? ModalConfirmationType.closeSet : ModalConfirmationType.openSet,
```

Comment:
is see you compare with status two times, probably it is better to create a common private method that will accept status and modal window type (this method will open/close modal window and then dispatch change status with proper status)
and pass a needed parameter from the outside, even from the template there can be two different buttons that are displayed according to the condition (it's up to you) 
![image](https://user-images.githubusercontent.com/27493959/184657042-5e54d92f-a0cb-4bed-af25-d05ee02eb06d.png)